src/steamship/agents/memory/chathistory.py

```diff
@@ -160,3 +161,24 @@ def messages(self) -> List[Block]:
     @property
     def client(self) -> Client:
         return self.file.client
+
+    def messages_as_string(
```

Comment:
this style suits me just as well as the other.  curious what the general preference is.

Reply:
@dkolas do you think we need/want/should have a `messages(strategy: Strategy, kwargs) -> List[Block]` and a separate `as_string(List[Block])` elsewhere?

I'm wondering if we will ever need to use a `Strategy` to retrieve blocks but do something different with them once retrieved.

Or, maybe, something like `ChatHistory.apply(visitor: HistoryVisitor)` ?  That would allow the `Strategy`/`Visitor`/`Whatever` to decide how to transform the messages as requested by the client?

Reply:
If you just want the blocks, you could just call my_selector.get_messages(chat_history), right?

Reply:
i suppose, but I guess that maybe feels like mixing approaches a bit for some strange reason to me.

i don't have strong feelings here, but I wonder if (and I can't believe I'm saying this) we had a method like `Block.as_context_string(...)` or something even in `Agent` _might_ serve us in the medium-ish term...

```python
selected_messages = agent.context.chat_history.messages(selector=buffered_window)
selected_history = agent.as_prompt_history(selected_messages)
```

As I type that out, I _almost_ think the format bits should (potentially) hang off the concept of a `Prompt` (if we ever introduce it). But maybe `Agent` is the right place for now?

Thoughts?  (apologies if this feels like yak shaving)